src/components/TopBar.tsx

```diff
@@ -0,0 +1,519 @@
+      <div
+        style={{
+          ...topBarStyle,
+          // This div contains the primary fallback color for a11y which will be used to calculate color contrast when the background is manually hidden
+          // A 1.47 intensity results in precisely a 4.5:1 ratio against the white text
+          backgroundColor: isTerra() ? colors.primary(1.47) : colors.light(1),
+        }}
+      >
+        <div
+          style={{
+            background: isTerra() ? `right url(${headerRightHexes})` : undefined,
+            backgroundRepeat: 'no-repeat',
+            flex: '1 1 auto',
+            display: 'flex',
+            alignSelf: 'stretch',
+            width: '100%',
+            alignItems: 'center',
+          }}
+        >
+          {
+            showMenu ? (
+              <Clickable
+                style={{
+                  alignSelf: 'stretch',
+                  display: 'flex',
+                  alignItems: 'center',
+                  padding: '0 1rem',
+                  margin: '2px 1rem 0 2px',
+                }}
+                onClick={navShown ? hideNav : showNav}
+                aria-expanded={navShown}
+                aria-label='Toggle main menu'
+              >
+                <Icon
+                  icon='bars'
+                  size={36}
+                  style={{
+                    color: isTerra() ? 'white' : colors.accent(),
+                    flex: 'none',
+                    transform: navShown ? 'rotate(90deg)' : undefined,
+                    transition: 'transform 0.1s ease-out',
+                  }}
+                />
+              </Clickable>
+            ) : (
+              <div style={{ width: 'calc(1rem + 1rem + 1rem + 2px + 36px)' }} />
+            ) // padding (l+r) + margin (l+r) + icon size
+          }
+          <a style={{ ...pageTitleStyles, display: 'flex', alignItems: 'center' }} href={href || Nav.getLink('root')}>
+            {topBarLogo()}
+            <div>{title && <h1 style={{ fontSize: '1em', fontWeight: 500, padding: 0, margin: 0 }}>{title}</h1>}</div>
+          </a>
+          <div style={{ display: 'flex', flexGrow: 1 }}>{children}</div>
+          <AlertsIndicator
+            style={{
+              margin: '0 1rem 0 0.5rem',
+              color: isTerra() ? 'white' : colors.dark(),
+            }}
+          />
+          {openFirecloudModal && <PreferFirecloudModal onDismiss={() => setOpenFirecloudModal(false)} />}
+        </div>
+      </div>
+      <RequiredUpdateAlert />
+      <SkipNavTarget ref={mainRef} />
+    </div>
+  );
+};
+
+const PreferFirecloudModal = ({ onDismiss }) => {
+  const [emailAgreed, setEmailAgreed] = useState<boolean>(true);
+  const [reason, setReason] = useState<string>('');
+  const [submitting, setSubmitting] = useState<boolean>(false);
+
+  const user = useStore(userStore);
+  const {
+    profile: { firstName, lastName },
+  } = user;
+  const email = user.profile.contactEmail ?? user.terraUser.email ?? '';
```

Comment:
I think this is the only place where I had to make a functional change to make the component work.
Before, this was getting the value from `userStore.profile.email`, which doesn't exist.

Reply:
And nobody should be using this anyway….